src/components/dialogs/contingency-list/explicit-naming/explicit-naming-form.js

```diff
@@ -20,15 +21,87 @@ import { gridItem } from '../../../utils/dialog-utils';
 import yup from '../../../utils/yup-config';
 import { makeDefaultRowData } from '../contingency-list-utils';
 import ChipsArrayEditor from '../../../utils/rhf-inputs/ag-grid-table-rhf/cell-editors/chips-array-editor';
+import { ContingencyListType } from 'utils/elementType';
+
+export const getExplicitNamingSchema = (id) => {
+    return {
+        [id]: yup
+            .array()
+            .of(
+                yup.object().shape({
+                    [CONTINGENCY_NAME]: yup.string().nullable(),
+                    [EQUIPMENT_IDS]: yup.array().of(yup.string().nullable()),
+                })
+            )
+            .when([CONTINGENCY_LIST_TYPE], {
+                is: ContingencyListType.EXPLICIT_NAMING.id,
+                then: (schema) =>
+                    schema
+                        .test(
+                            'rowWithoutNameOrEquipments',
+                            'contingencyTablePartiallyDefinedError',
+                            (array) => {
+                                return !array.some(
+                                    (row) =>
+                                        (!row[CONTINGENCY_NAME]?.trim() &&
+                                            row[EQUIPMENT_IDS]?.length) ||
+                                        (row[CONTINGENCY_NAME]?.trim() &&
+                                            !row[EQUIPMENT_IDS]?.length)
```

Comment:
you can extract method here and use it for getExplicitNamingSchema and getExplicitNamingEditSchema

Reply:
Done